File: TransCoder_tokenized_test_set_functions/cpp/CHANGE_ARRAY_PERMUTATION_NUMBERS_1_N.cpp
void makePermutation ( int a [ ], int n ) {
  unordered_map < int, int > count;
  for ( int i = 0;
  i < n;
  i ++ ) count [ a [ i ] ] ++;
  int next_missing = 1;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( count [ a [ i ] ] != 1 || a [ i ] > n || a [ i ] < 1 ) {
      count [ a [ i ] ] --;
      while ( count . find ( next_missing ) != count . end ( ) ) next_missing ++;
      a [ i ] = next_missing;
      count [ next_missing ] = 1;
    }
  }
}

File: TransCoder_tokenized_test_set_functions/java/CHANGE_ARRAY_PERMUTATION_NUMBERS_1_N.java
static void makePermutation ( int [ ] a, int n ) {
  HashMap < Integer, Integer > count = new HashMap < Integer, Integer > ( );
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( count . containsKey ( a [ i ] ) ) {
      count . put ( a [ i ], count . get ( a [ i ] ) + 1 );
    }
    else {
      count . put ( a [ i ], 1 );
    }
  }
  int next_missing = 1;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( count . containsKey ( a [ i ] ) && count . get ( a [ i ] ) != 1 || a [ i ] > n || a [ i ] < 1 ) {
      count . put ( a [ i ], count . get ( a [ i ] ) - 1 );
      while ( count . containsKey ( next_missing ) ) next_missing ++;
      a [ i ] = next_missing;
      count . put ( next_missing, 1 );
    }
  }
}